decksite/data/rule.py

```diff
@@ -0,0 +1,189 @@
+from typing import List
+
+from decksite.data import deck
+from decksite.database import db
+from magic.models import Deck
+from shared.container import Container
+
+
+def num_classified_decks() -> int:
+    sql = 'SELECT COUNT(DISTINCT(deck_id)) AS c FROM ({apply_rules_query}) AS applied_rules'.format(apply_rules_query=apply_rules_query())
+    return (db().select(sql))[0]['c']
+
+def mistagged_decks() -> List[Deck]:
+    sql = """SELECT
+                    deck_id, rule_archetype.name AS rule_archetype_name, tagged_archetype.name AS tagged_archetype_name
+                FROM
+                    ({apply_rules_query}) AS applied_rules
+                JOIN
+                    deck
+                ON
+                    applied_rules.deck_id = deck.id
+                JOIN
+                    archetype AS rule_archetype
+                ON
+                    rule_archetype.id = applied_rules.archetype_id
+                JOIN
+                    archetype AS tagged_archetype
+                ON
+                    tagged_archetype.id = deck.archetype_id
+                WHERE
+                    rule_archetype.id != tagged_archetype.id""".format(apply_rules_query=apply_rules_query())
```

Comment:
nit. Standard is to put the closing triple quotes on the line below here. I don't know if we should have standards that are not lint rules but I also don't like writing lint rules either :D